iDiff/differs/history.go

```diff
@@ -8,31 +8,45 @@ import (
 	"strings"
 
 	"github.com/GoogleCloudPlatform/runtimes-common/iDiff/utils"
-
+	"github.com/docker/docker/api/types/image"
 	"github.com/docker/docker/client"
+	"github.com/golang/glog"
 	"golang.org/x/net/context"
 )
 
 // History compares the Docker history for each image.
-func History(img1, img2 string, json bool) (string, error) {
-	return getHistoryDiff(img1, img2, json)
+func History(img1, img2 string, json bool, eng bool) (string, error) {
+	return getHistoryDiff(img1, img2, json, eng)
 }
 
-func getHistoryList(image string) ([]string, error) {
-	ctx := context.Background()
-	cli, err := client.NewEnvClient()
+func getHistoryList(img string, eng bool) ([]string, error) {
+	validDocker, err := utils.ValidDockerVersion(eng)
 	if err != nil {
 		return []string{}, err
 	}
-	history, err := cli.ImageHistory(ctx, image)
-	if err != nil {
-		return []string{}, err
+	var history []image.HistoryResponseItem
+	if validDocker {
+		ctx := context.Background()
+		cli, err := client.NewEnvClient()
+		if err != nil {
+			return []string{}, err
+		}
+		history, err = cli.ImageHistory(ctx, img)
+		if err != nil {
+			return []string{}, err
+		}
+	} else {
+		glog.Info("Docker version incompatible with api, shelling out to local Docker client.")
+		history, err = utils.GetImageHistory(img)
```

Comment:
If you set the client API version appropriately for the server, you won't have to shell out